### ISSUE TITLE
[Bug]: Latest SD.Next version breaks Adetailer

### ISSUE THREAD

**Issue by demib72** [Bing-su/adetailer#403]
### Describe the bug

Recently updated to the most recent version of SD.Next. Was working fine until I both updated to the latest SD.Next and updated the extensions I have installed. Initially had a problem Height not being over 0 since I was using a non-standard size (602x899). But then I started getting this issue any time I use CN's tile processor. 

### Console logs, from start to end.

```Shell
username@pop-os:~/Applications/automatic$ ./webui.sh --medvram
Create and activate python venv
Launching launch.py...
21:50:00-786061 INFO     Starting SD.Next                                                                 
21:50:00-788840 INFO     Python 3.10.12 on Linux                                                          
21:50:00-833847 INFO     Version: app=sd.next updated=2023-11-03 hash=dc79cec1                            
                         url=https://github.com/vladmandic/automatic.git/tree/master                      
21:50:02-316669 INFO     Platform: arch=x86_64 cpu=x86_64 system=Linux release=6.5.6-76060506-generic     
                         python=3.10.12                                                                   
21:50:02-318100 INFO     nVidia CUDA toolkit detected: nvidia-smi present                                 
21:50:02-343419 WARNING  Modified files: ['webui.sh']                                                     
21:50:02-356020 INFO     Extensions: disabled=[]                                                          
21:50:02-357450 INFO     Extensions: enabled=['stable-diffusion-webui-rembg',                             
                         'stable-diffusion-webui-images-browser', 'sd-webui-controlnet',                  
                         'multidiffusion-upscaler-for-automatic1111', 'clip-interrogator-ext',            
                         'a1111-sd-webui-lycoris', 'sd-extension-system-info', 'ScuNET', 'SwinIR',        
                         'sd-extension-chainner', 'sd-webui-agent-scheduler', 'Lora', 'LDSR',             
                         'sd-dynamic-thresholding'] extensions-builtin                                    
21:50:02-361266 INFO     Extensions: enabled=['openpose-editor', 'sd-webui-reactor',                      
                         'sd-webui-openpose-editor', 'adetailer'] extensions                              
21:50:02-363510 INFO     No changes detected: quick launch active                                         
21:50:02-364611 INFO     Verifying requirements                                                           
21:50:02-389361 INFO     Verifying packages                                                               
21:50:02-392169 INFO     Extensions: disabled=[]                                                          
21:50:02-393378 INFO     Extensions: enabled=['stable-diffusion-webui-rembg',                             
                         'stable-diffusion-webui-images-browser', 'sd-webui-controlnet',                  
                         'multidiffusion-upscaler-for-automatic1111', 'clip-interrogator-ext',            
                         'a1111-sd-webui-lycoris', 'sd-extension-system-info', 'ScuNET', 'SwinIR',        
                         'sd-extension-chainner', 'sd-webui-agent-scheduler', 'Lora', 'LDSR',             
                         'sd-dynamic-thresholding'] extensions-builtin                                    
21:50:02-396439 INFO     Extensions: enabled=['openpose-editor', 'sd-webui-reactor',                      
                         'sd-webui-openpose-editor', 'adetailer'] extensions                              
21:50:02-401881 INFO     Extension preload: {'extensions-builtin': 0.0, 'extensions': 0.0}                
21:50:02-403859 INFO     Command line args: ['--medvram'] medvram=True                                    
21:50:06-662625 INFO     Engine: backend=Backend.ORIGINAL compute=cuda mode=no_grad device=cuda           
                         cross-optimization="Scaled-Dot-Product"                                          
21:50:06-784341 INFO     Device: device=NVIDIA GeForce RTX 2060 n=1 arch=sm_90 cap=(7, 5) cuda=11.8       
                         cudnn=8700 driver=535.113.01                                                     
21:50:07-341786 INFO     Available VAEs: path="/home/username/Applications/automatic/models/VAE" items=4 
21:50:07-372472 INFO     Available models:                                                                
                         path="/home/username/Applications/automatic/models/Stable-diffusion" items=13   
                         time=0.03s                                                                       
21:50:08-227285 INFO     Extension:                                                                       
                         script='extensions-builtin/sd-dynamic-thresholding/scripts/dynamic_thresholding.p
                         y' ======                                                                        
21:50:08-228453 INFO     Extension:                                                                       
                         script='extensions-builtin/sd-dynamic-thresholding/scripts/dynamic_thresholding.p
                         y' Error! UniPC sampler support failed to load! Is your WebUI up to date?        
21:50:08-229296 INFO     Extension:                                                                       
                         script='extensions-builtin/sd-dynamic-thresholding/scripts/dynamic_thresholding.p
                         y' (Error: module 'modules.models.diffusion.uni_pc' has no attribute 'sampler')  
21:50:08-230356 INFO     Extension:                                                                       
                         script='extensions-builtin/sd-dynamic-thresholding/scripts/dynamic_thresholding.p
                         y' ======                                                                        
21:50:08-669370 INFO     Extension:                                                                       
                         script='extensions-builtin/sd-webui-agent-scheduler/scripts/task_scheduler.py'   
                         Using sqlite file:                                                               
                         extensions-builtin/sd-webui-agent-scheduler/task_scheduler.sqlite3               
21:50:08-952169 INFO     Extension: script='extensions-builtin/sd-webui-controlnet/scripts/controlnet.py' 
                         Warning: ControlNet failed to load SGM - will use LDM instead.                   
21:50:08-953231 INFO     Extension: script='extensions-builtin/sd-webui-controlnet/scripts/controlnet.py' 
                         ControlNet preprocessor location:                                                
                         /home/username/Applications/automatic/extensions-builtin/sd-webui-controlnet/ann
                         otator/downloads                                                                 
21:50:08-958374 INFO     Extension: script='extensions-builtin/sd-webui-controlnet/scripts/hook.py'       
                         Warning: ControlNet failed to load SGM - will use LDM instead.                   
21:50:13-169920 INFO     Extension: script='extensions/adetailer/scripts/!adetailer.py' [-] ADetailer     
                         initialized. version: 23.8.1, num models: 9                                      
21:50:13-426717 STATUS   Extensions time: 6.03s { a1111-sd-webui-lycoris=0.53s Lora=0.20s                 
                         sd-dynamic-thresholding=0.05s sd-extension-chainner=0.06s                        
                         sd-webui-agent-scheduler=0.33s sd-webui-controlnet=0.29s                         
                         stable-diffusion-webui-images-browser=0.09s stable-diffusion-webui-rembg=0.50s   
                         adetailer=3.62s sd-webui-reactor=0.17s }                                         
21:50:13-629918 STATUS   No internet access detected, using default fonts                                 
21:50:13-631440 STATUS   Loading UI theme: name=derekzen/stardust style=Dark base=base.css                
21:50:13-632122 WARNING  Using 3rd party theme which is not optimized for SD.Next                         
21:50:14-981117 ERROR    Theme download error accessing HuggingFace                                       
21:50:16-648472 STATUS   Local URL: http://127.0.0.1:7860/                                                
21:50:16-649429 STATUS   Initializing middleware                                                          
21:50:16-838224 STATUS   [AgentScheduler] Task queue is empty                                             
21:50:16-839216 STATUS   [AgentScheduler] Registering APIs                                                
21:50:17-256119 STATUS   Select: model="epicrealism_pureEvolutionV5 [76be5be1b2]"                         
Loading weights: /home/username/Applications/automatic/models/Stable-diffusion/epicrealism_pureEvolutio… 
21:50:17-396013 STATUS   Torch override VAE dtype: no-half set                                            
21:50:17-397012 STATUS   Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float32      
                         unet=torch.float16 context=no_grad fp16=True bf16=False                          
21:50:17-843051 STATUS   LDM: LatentDiffusion: Running in eps-prediction mode                             
21:50:17-844036 STATUS   LDM: DiffusionWrapper has 859.52 M params.                                       
21:50:17-844776 STATUS   Diffusers: model="Stable Diffusion" class=StableDiffusionPipeline                
                         file="/home/username/Applications/automatic/models/Stable-diffusion/epicrealism_
                         pureEvolutionV5.safetensors" size=2034MB                                         
21:50:18-726358 STATUS   Cross-attention: optimization=Scaled-Dot-Product options=[]                      
21:50:18-753429 STATUS   Loaded embeddings: loaded=32 skipped=0 time=0.02s                                
21:50:18-759430 STATUS   Model loaded in 1.50s { load=0.12s create=0.44s apply=0.55s vae=0.33s }          
21:50:19-059089 STATUS   Model load finished: {'ram': {'used': 7.29, 'total': 31.23}, 'gpu': {'used': 0.8,
                         'total': 5.78}, 'retries': 0, 'oom': 0} cached=0                                 
21:50:19-150687 STATUS   Startup time: 16.74s { torch=3.10s gradio=0.58s libraries=1.23s extensions=6.03s 
                         upscalers=0.10s ui-extra-networks=1.61s ui-txt2img=0.10s ui-img2img=0.13s        
                         ui-settings=0.33s ui-extensions=0.56s ui-defaults=0.07s launch=0.26s api=0.08s   
                         app-started=0.53s checkpoint=1.89s }                                             
100%|█████████████████████████████████████████████████████████████████████| 30/30 [00:20<00:00,  1.44it/s]
21:51:41-566451 ERROR    Running script postprocess image: extensions/adetailer/scripts/!adetailer.py:    
                         AttributeError                                                                   
╭────────────────────────────────── Traceback (most recent call last) ───────────────────────────────────╮
│ /home/username/Applications/automatic/modules/scripts.py:553 in postprocess_image                     │
│                                                                                                        │
│   552 │   │   │   │   args = p.per_script_args.get(script.title(), p.script_args[script.args_f         │
│ ❱ 553 │   │   │   │   script.postprocess_image(p, pp, *args)                                           │
│   554 │   │   │   except Exception as e:                                                               │
│                                                                                                        │
│ /home/username/Applications/automatic/extensions/adetailer/adetailer/traceback.py:159 in wrapper      │
│                                                                                                        │
│   158 │   │   │   │   error = RuntimeError(output)                                                     │
│ ❱ 159 │   │   │   raise error from None                                                                │
│   160                                                                                                  │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 
╭────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                            System info                                             │
│ ┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓ │
│ ┃             ┃ Value                                                                            ┃ │
│ ┡━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩ │
│ │    Platform │ Linux-6.5.6-76060506-generic-x86_64-with-glibc2.35                               │ │
│ │      Python │ 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]                               │ │
│ │     Version │ Unknown (too old or vladmandic)                                                  │ │
│ │      Commit │ Unknown                                                                          │ │
│ │ Commandline │ ['launch.py', '--medvram']                                                       │ │
│ │   Libraries │ {'torch': '2.0.1+cu118', 'torchvision': '0.15.2+cu118', 'ultralytics':           │ │
│ │             │ '8.0.164', 'mediapipe': '0.10.3'}                                                │ │
│ └─────────────┴──────────────────────────────────────────────────────────────────────────────────┘ │
│                                               Inputs                                               │
│ ┏━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓ │
│ ┃                 ┃ Value                                                                        ┃ │
│ ┡━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩ │
│ │          prompt │ (highly detailed:1.2),(light grey background), reflections, refractions,     │ │
│ │                 │ (realistic:1.3), (full body:1.3), full body, 1girl, reference sheet, a       │ │
│ │                 │ woman, turnaround, detailed skin + face + hair + eyes, (skin pores:0.6),     │ │
│ │                 │ depth of field, photographed on a Canon EOS, intricate details, (sharp       │ │
│ │                 │ focus)                                                                       │ │
│ │ negative_prompt │ CyberRealistic_Negative-neg, bad-hands-3, curly hair, sandals, 3dcg, unreal, │ │
│ │                 │ (close-up:1.3), paintings, sketches, (worst quality:2), (low quality:2),     │ │
│ │                 │ (normal quality:2), lowres, normal quality, ((monochrome)), ((grayscale)),   │ │
│ │                 │ anime, cartoon, drawing, illustration, boring, long neck, out of frame,      │ │
│ │                 │ extra fingers, mutated hands, monochrome, ((poorly drawn hands)), ((poorly   │ │
│ │                 │ drawn face)), (((mutation))), (((deformed))), ((ugly)), blurry, ((bad        │ │
│ │                 │ anatomy)), (((bad proportions))), ((extra limbs)), cloned face, glitchy,     │ │
│ │                 │ bokeh, (((long neck))), red eyes, extra heads, close up, text ,watermarks,   │ │
│ │                 │ logo3                                                                        │ │
│ │          n_iter │ 1                                                                            │ │
│ │      batch_size │ 1                                                                            │ │
│ │           width │ 640                                                                          │ │
│ │          height │ 960                                                                          │ │
│ │    sampler_name │ DPM++ 2M SDE                                                                 │ │
│ │       enable_hr │ False                                                                        │ │
│ │     hr_upscaler │ None                                                                         │ │
│ │      checkpoint │ epicrealism_pureEvolutionV5 [76be5be1b2]                                     │ │
│ │             vae │ Automatic                                                                    │ │
│ │            unet │ ------                                                                       │ │
│ └─────────────────┴──────────────────────────────────────────────────────────────────────────────┘ │
│                      ADetailer                                                                     │
│ ┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                                                │
│ ┃                     ┃ Value                     ┃                                                │
│ ┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩                                                │
│ │             version │ 23.8.1                    │                                                │
│ │            ad_model │ face_yolov8n.pt           │                                                │
│ │           ad_prompt │                           │                                                │
│ │  ad_negative_prompt │                           │                                                │
│ │ ad_controlnet_model │ control_v11f1e_sd15_tile  │                                                │
│ │              is_api │ False                     │                                                │
│ └─────────────────────┴───────────────────────────┘                                                │
│ ╭────────────────────────────── Traceback (most recent call last) ───────────────────────────────╮ │
│ │ /home/username/Applications/automatic/extensions/adetailer/adetailer/traceback.py:139 in      │ │
│ │ wrapper                                                                                        │ │
│ │                                                                                                │ │
│ │   136 │   │   width = width - 4 if width > 4 else None                                         │ │
│ │   137 │   │   console = Console(file=string, width=width)                                      │ │
│ │   138 │   │   try:                                                                             │ │
│ │ ❱ 139 │   │   │   return func(*args, **kwargs)                                                 │ │
│ │   140 │   │   except Exception as e:                                                           │ │
│ │   141 │   │   │   tables = [                                                                   │ │
│ │   142 │   │   │   │   get_table(title, data)                                                   │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions/adetailer/scripts/!adetailer.py:639 in       │ │
│ │ postprocess_image                                                                              │ │
│ │                                                                                                │ │
│ │   636 │   │   │   for n, args in enumerate(arg_list):                                          │ │
│ │   637 │   │   │   │   if args.ad_model == "None":                                              │ │
│ │   638 │   │   │   │   │   continue                                                             │ │
│ │ ❱ 639 │   │   │   │   is_processed |= self._postprocess_image(p, pp, args, n=n)                │ │
│ │   640 │   │                                                                                    │ │
│ │   641 │   │   if is_processed:                                                                 │ │
│ │   642 │   │   │   self.save_image(                                                             │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions/adetailer/scripts/!adetailer.py:546 in       │ │
│ │ _postprocess_image                                                                             │ │
│ │                                                                                                │ │
│ │   543 │   │                                                                                    │ │
│ │   544 │   │   i = p._ad_idx                                                                    │ │
│ │   545 │   │                                                                                    │ │
│ │ ❱ 546 │   │   i2i = self.get_i2i_p(p, args, pp.image)                                          │ │
│ │   547 │   │   seed, subseed = self.get_seed(p)                                                 │ │
│ │   548 │   │   ad_prompts, ad_negatives = self.get_prompt(p, args)                              │ │
│ │   549                                                                                          │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions/adetailer/scripts/!adetailer.py:429 in       │ │
│ │ get_i2i_p                                                                                      │ │
│ │                                                                                                │ │
│ │   426 │   │   i2i._disable_adetailer = True                                                    │ │
│ │   427 │   │                                                                                    │ │
│ │   428 │   │   if args.ad_controlnet_model != "None":                                           │ │
│ │ ❱ 429 │   │   │   self.update_controlnet_args(i2i, args)                                       │ │
│ │   430 │   │   else:                                                                            │ │
│ │   431 │   │   │   i2i.control_net_enabled = False                                              │ │
│ │   432                                                                                          │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions/adetailer/scripts/!adetailer.py:160 in       │ │
│ │ update_controlnet_args                                                                         │ │
│ │                                                                                                │ │
│ │   157 │   │   │   and self.controlnet_ext.cn_available                                         │ │
│ │   158 │   │   │   and args.ad_controlnet_model != "None"                                       │ │
│ │   159 │   │   ):                                                                               │ │
│ │ ❱ 160 │   │   │   self.controlnet_ext.update_scripts_args(                                     │ │
│ │   161 │   │   │   │   p,                                                                       │ │
│ │   162 │   │   │   │   model=args.ad_controlnet_model,                                          │ │
│ │   163 │   │   │   │   module=args.ad_controlnet_module,                                        │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions/adetailer/controlnet_ext/controlnet_ext.py:8 │ │
│ │ 1 in update_scripts_args                                                                       │ │
│ │                                                                                                │ │
│ │    78 │   │   │   )                                                                            │ │
│ │    79 │   │   ]                                                                                │ │
│ │    80 │   │                                                                                    │ │
│ │ ❱  81 │   │   self.external_cn.update_cn_script_in_processing(p, cn_units)                     │ │
│ │    82                                                                                          │ │
│ │    83                                                                                          │ │
│ │    84 def get_cn_model_dirs() -> list[Path]:                                                   │ │
│ │                                                                                                │ │
│ │ /home/username/Applications/automatic/extensions-builtin/sd-webui-controlnet/internal_control │ │
│ │ net/external_code.py:339 in update_cn_script_in_processing                                     │ │
│ │                                                                                                │ │
│ │   336 │   - ControlNet is not present in `p.scripts`                                           │ │
│ │   337 │   - `p.script_args` is not filled with script arguments for scripts that are processed │ │
│ │   338 │   """                                                                                  │ │
│ │ ❱ 339 │   p.script_args = update_cn_script(p.scripts, p.script_args_value, cn_units)           │ │
│ │   340                                                                                          │ │
│ │   341                                                                                          │ │
│ │   342 def update_cn_script(                                                                    │ │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────╯ │
│ AttributeError: 'StableDiffusionProcessingImg2Img' object has no attribute 'script_args_value'     │
╰────────────────────────────────────────────────────────────────────────────────────────────────────╯

21:51:41-866171 STATUS   Processed: images=1 time=35.79s its=0.84 memory={'ram': {'used': 6.42, 'total':  
                         31.23}, 'gpu': {'used': 2.67, 'total': 5.78}, 'retries': 0, 'oom': 0}
```

**Comment by demib72** [comment]
I see the source of the problem, it is a controlnet issue (https://github.com/Mikubill/sd-webui-controlnet/commit/f2aafcf2beb99a03cbdf7db73852228ccd6bd1d6). Extra code was added to the external_code file to fix another issue.

**Comment by demib72** [comment]
I figured the problem, in automatic1111 on line 162 of modules/processing.py the class value is script_args_value. But in SD.Next implementation of the file it is simply script_args on line 156. Change line 339 controlnet's external_code.py to match whatever webui is using fixes it for me.

**Comment by Aptronymist** [comment]
We're looking at it on our end, but vlad didn't think he had changed it recently.

**Comment by brknsoul** [comment]
Ensure you've updated ControlNet and Adetailer.
I'm using the latest SD.Next (a2b50b69), ControlNet (fce6775a), and Adetailer (6b41b3db) and I'm encountering no issues.

**Comment by demib72** [comment]
@Aptronymist Thanks, it is funny how the simplest thing can cause so much frustration in programming.

**Comment by demib72** [comment]
@brknsoul Reinstalled SD.Next to latest (a2b50b69), hard reset Controlnet to fce6775a, and latest Adetailer (6b41b3d). Doesn't work (RuntimeError: [-] Adetailer: ControlNet option not available in WEBUI version lower than 1.6.0 due to updates in ControlNet). Makes sense, Adetailer wants A1111.

Edited the files a bit to apply my bandaid. Other then it squeaking [tile_resample.threshold_a] Invalid value(-1), using default value 1.0 due to enhancements in adetailer, it works enough for me.

I await the real fix. :)

**Comment by Bing-su** [comment]
This is because ControlNet is referencing an attribute (`script_args_value`) that only exists on the A1111.

This should be fixed in ControlNet or SD.Next, not by me.
